File: docs/tutorial/go/client/users_service.go
package main

import (
	"encoding/json"
	"net/http"
)

type UsersService service

// Get list of all developers
func (s *UsersService) UsersGet(headers, queryParams map[string]interface{}) ([]User, *http.Response, error) {
	var u []User

	resp, err := s.client.doReqNoBody("GET", s.client.BaseURI+"/users", headers, queryParams)
	if err != nil {
		return u, nil, err
	}
	defer resp.Body.Close()

	return u, resp, json.NewDecoder(resp.Body).Decode(&u)
}

// Add user
func (s *UsersService) UsersPost(user User, headers, queryParams map[string]interface{}) (User, *http.Response, error) {
	var u User

	resp, err := s.client.doReqWithBody("POST", s.client.BaseURI+"/users", &user, headers, queryParams)
	if err != nil {
		return u, nil, err
	}
	defer resp.Body.Close()

	return u, resp, json.NewDecoder(resp.Body).Decode(&u)
}

// Get information on a specific user
func (s *UsersService) UsersUsernameGet(username string, headers, queryParams map[string]interface{}) (User, *http.Response, error) {
	var u User

	resp, err := s.client.doReqNoBody("GET", s.client.BaseURI+"/users/"+username, headers, queryParams)
	if err != nil {
		return u, nil, err
	}
	defer resp.Body.Close()

	return u, resp, json.NewDecoder(resp.Body).Decode(&u)
}

File: docs/tutorial/python/aiohttp_client/goramldir/users_service.py
class UsersService:
    def __init__(self, client):
        self.client = client



    async def users_get(self, headers=None, query_params=None):
        """
        Get list of all developers
        It is method for GET /users
        """
        uri = self.client.base_url + "/users"
        return await self.client.get(uri, headers=headers, params=query_params)


    async def users_post(self, data, headers=None, query_params=None):
        """
        Add user
        It is method for POST /users
        """
        uri = self.client.base_url + "/users"
        return await self.client.post(uri, data, headers=headers, params=query_params)


    async def users_byUsername_get(self, username, headers=None, query_params=None):
        """
        Get information on a specific user
        It is method for GET /users/{username}
        """
        uri = self.client.base_url + "/users/"+username
        return await self.client.get(uri, headers=headers, params=query_params)
